components/brave_ads/core/internal/client/ads_client_util.cc

```diff
@@ -0,0 +1,456 @@
+}
+
+void ClearProfilePref(const std::string& path) {
+  GetInstance()->ClearProfilePref(path);
+}
+
+bool HasProfilePrefPath(const std::string& path) {
+  return GetInstance()->HasProfilePrefPath(path);
+}
+
+absl::optional<base::Value> GetLocalStatePref(const std::string& path) {
+  return GetInstance()->GetLocalStatePref(path);
+}
+
+bool GetLocalStateBooleanPref(const std::string& path) {
+  const absl::optional<base::Value> value = GetLocalStatePref(path);
+  CHECK(value) << "GetLocalStateBooleanPref failed: " << path;
+  CHECK(value->is_bool()) << "Wrong type for GetProfileBooleanPref: " << path;
+
+  return value->GetBool();
+}
+
+int GetLocalStateIntegerPref(const std::string& path) {
+  const absl::optional<base::Value> value = GetLocalStatePref(path);
+  CHECK(value) << "GetLocalStateIntegerPref failed: " << path;
+  CHECK(value->is_int()) << "Wrong type for GetProfileBooleanPref: " << path;
+
+  return value->GetInt();
+}
+
+double GetLocalStateDoublePref(const std::string& path) {
+  const absl::optional<base::Value> value = GetLocalStatePref(path);
+  CHECK(value) << "GetLocalStateDoublePref failed: " << path;
+  CHECK(value->is_double()) << "Wrong type for GetProfileBooleanPref: " << path;
+
+  return value->GetDouble();
+}
+
+std::string GetLocalStateStringPref(const std::string& path) {
+  const absl::optional<base::Value> value = GetLocalStatePref(path);
+  CHECK(value) << "GetLocalStateStringPref failed: " << path;
+  CHECK(value->is_string()) << "Wrong type for GetProfileBooleanPref: " << path;
+
+  return value->GetString();
+}
+
+base::Value::Dict GetLocalStateDictPref(const std::string& path) {
+  const absl::optional<base::Value> value = GetLocalStatePref(path);
+  CHECK(value) << "GetLocalStateDictPref failed: " << path;
+  CHECK(value->is_dict()) << "Wrong type for GetLocalStateDictPref: " << path;
+
+  return value->GetDict().Clone();
+}
+
+base::Value::List GetLocalStateListPref(const std::string& path) {
+  const absl::optional<base::Value> value = GetLocalStatePref(path);
+  CHECK(value) << "GetLocalStateListPref failed: " << path;
+  CHECK(value->is_list()) << "Wrong type for GetLocalStateListPref: " << path;
+
+  return value->GetList().Clone();
+}
+
+int64_t GetLocalStateInt64Pref(const std::string& path) {
+  const absl::optional<base::Value> value = GetLocalStatePref(path);
+  CHECK(value) << "GetLocalStateInt64Pref failed: " << path;
+  CHECK(value->is_string()) << "Wrong type for GetProfileBooleanPref: " << path;
+
+  const absl::optional<int64_t> integer = base::ValueToInt64(*value);
+  return integer.value_or(0);
+}
+
+uint64_t GetLocalStateUint64Pref(const std::string& path) {
+  const absl::optional<base::Value> value = GetLocalStatePref(path);
+  CHECK(value) << "GetLocalStateUint64Pref failed: " << path;
+  CHECK(value->is_string()) << "Wrong type for GetProfileBooleanPref: " << path;
+
+  uint64_t integer;
+  base::StringToUint64(value->GetString(), &integer);
```

Comment:
should we return 0 if the conversion failed?

Reply:
Agreed amd copied Chromium where I have added DCHECK's, so that we catch any potential bugs